WordPress/gutenberg
# [RNMobile] Enable Audio Block

## Description


This PR enables Audio Block for mobile.

GB Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3070


## How has this been tested?




### Test upload:

1. Add Audio block
1. Choose to upload:
	* From device
	* From WP Media Library

### Test replace:

1. Add Audio block
1. Choose an audio
1. Press audio block
1. Select "Replace audio" from the toolbar
1. Choose a new audio

### Test caption:

1. Add Audio block
1. Choose an audio
1. The block should be selected with "Write caption..." placeholder for caption field
1. De-select block
1. Caption field should be hidden
1. Switch to HTML
1. Check there are no `<figcaption>` tags
1. Switch to Visual
1. Select Audio block
1. Type some words in the caption field
1. Add some bold/italic words
1. Move the cursor to the middle and press enter
1. Caption should have a new line
1. Move the cursor to the end and press enter
1. A new paragraph block should be created
1. Switch to HTML
1. Check if `<figcaption>` tags contain correct information (for bold, italic, etc. and `<br>` tags for new lines)
1. Save the post and open it from the web browser
1. Check if the caption shows correctly 


### Test settings:

1. Add Audio block
1. Choose audio
1. Press gear icon and modify settings
1. Check if they are reflected by previewing the post

### Test playing:

1. Add Audio block
1. Choose an audio
1. Tap "OPEN" to play the audio

### Test cancel action:

* Add a Audio block to an empty new post.
* Add a large file to the block.
* While the file is uploading, tap on theAudio  block to see the cancel action sheet/dialog.
    * Tap the Stop upload action.
    * The Audio block should go back to the empty state.
    
### Test error and retry:

* Add a Audio block to an empty new post.
* While it's uploading, turn un Airplane mode.
    * See the Audio block switch to its error state.
* While in the error state, tap on the Audio block to see the Retry action sheet/dialog.
* After reconnecting the device, press the Retry action.
    * Check that the audio file starts uploading again until it successfully finishes uploading.


## Screenshots 

<img width="300" alt="audio-block-states" src="https://user-images.githubusercontent.com/1845482/106293699-96afb200-624e-11eb-83fa-cd01cb229fe1.png">

## Types of changes


New feature (non-breaking change which adds functionality)




## Checklist:
